src/ui/public/vis/vis.js

```diff
@@ -206,17 +206,27 @@ export function VisProvider(Private, indexPatterns, getAppState) {
     }
 
     getCurrentState(includeDisabled) {
+      return this.getSerializableState(this, includeDisabled);
+    }
+
+    getSerializableState(state, includeDisabled) {
       return {
-        title: this.title,
-        type: this.type.name,
-        params: this.params,
-        aggs: this.aggs
+        title: state.title,
+        type: state.type.name || state.type,
+        params: _.cloneDeep(state.params),
+        aggs: state.aggs
           .map(agg => agg.toJSON())
           .filter(agg => includeDisabled || agg.enabled)
           .filter(Boolean)
       };
     }
 
+    copyCurrentState(includeDisabled) {
```

Comment:
I would add `includeDisabled = false` just to make it visible directly when seeing the method what's the default value of this, instead of needing to search which method is actually called by this one, and what that method does with the parameter. (Maybe also give it a default value in `getSerializableState`.)